config/conf.go

```diff
@@ -1,105 +1,127 @@
 package config
 
 import (
-	"encoding/json"
-	"errors"
 	"fmt"
+	"strings"
 
-	"github.com/xeipuuv/gojsonschema"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
 )
 
-// CFGLoader defines an interface for loading
-// config files.
-type CFGLoader interface {
-	Load(v any) error
-}
-
-// Config is the base definition of a Config file.
-// It contains config Data retrieved and unmarshaled
-// by a CFGLoader.
-type Config struct {
-	// Data is unmarshaled config data.
-	Data map[string]any
-	// Loaded signifies whether Data has already been
-	// loaded by Loader. All Config methods check this to
-	// see if Data needs to be re-loaded before use.
-	Loaded bool
-	// Loader retrieves and unmarshals the config data.
-	Loader CFGLoader
-}
-
-// NewConfig returns a new config set with the passed in cfgLoader.
-func NewConfig(cfgLoader *Loader) *Config {
-	res := &Config{
-		Loader: cfgLoader,
+// A map that maps config keys i.e. "cluster.host" to flag names i.e. "host".
+// This is only needed because if "instance". Otherwise we would just run
+// RegisterAlias inside the BindPFlags function.
+var configToFlagMap = map[string]string{}
+
+// InitConfig reads in config file and ENV variables if set. Should be called
+// from the root commands PersistentPreRunE function with the flags of the current command.
+func InitConfig(cfgFile, instance string, flags *pflag.FlagSet) (string, error) {
+	if cfgFile != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		viper.AddConfigPath(".")
+		viper.AddConfigPath(AsToolsConfDir)
+		viper.SetConfigName(AsToolsConfName)
 	}
 
-	return res
-}
+	var configFileUsed string
 
-// Load loads the config data into the Config.Data.
-func (o *Config) Load() error {
-	if o.Loaded {
-		return nil
+	if strings.HasSuffix(cfgFile, ".conf") {
+		// If .conf then explicitly set type to toml.
+		viper.SetConfigType("toml")
```

Comment:
If `cfgFile` is an empty string, then the default config path and name is used, which ends in ".conf". Since `cfgFile` is empty won't it miss the if condition where config type is set to toml? Will that cause an issue?

Reply:
ah, good catch. I actually changed this recently and couldn't understand why I had it differently before